R/epidist_utils.R

```diff
@@ -0,0 +1,703 @@
+#' Specify distribution parameter uncertainty
+#'
+#' @description A helper function when creating uncertainty for the parameters
+#' of the distribution for the `epidist` object
+#'
+#' @param ci_limits A numeric vector of length two with the lower and upper bound of
+#' the confidence interval or credible interval
+#' @param ci A numeric specifying the interval for the ci, e.g. 95 is
+#' 95% ci
+#' @param ci_type A character string, either "confidence interval" or "credible
+#' interval"
+#'
+#' @return List of three elements
+#' @export
+#'
+#' @examples
+#' # example with uncertainty for a single parameter
+#' create_epidist_uncertainty(
+#'   ci_limits = c(1, 3),
+#'   ci = 95,
+#'   ci_type = "confidence interval"
+#' )
+#'
+#' # example for multiple parameters
+#' # lengh of list should match number of parameters
+#' list(
```

Comment:
This way of specifying uncertainty implies that the uncertainty in parameters is uncorrelated, which it normally won't be - if that's how uncertainty is reported then obviously there is nothing that can be done about it but perhaps it would be useful to think about how it could be incorporated if it is accessible (or if the joint distribution can be estimated from raw data)?

Reply:
I had overlooked this aspect. I think most papers report uncertainty without specifying a correlation between them. What structure would be best for this, a n-dim matrix with n parameters?

Reply:
Issue opened for further work (#101).